webapp/loggedin/surveyViews/nodeDefsSelector/components/attributesSelector.js

```diff
@@ -1,83 +1,77 @@
 import React from 'react'
-import { connect } from 'react-redux'
+import * as PropTypes from 'prop-types'
 import * as R from 'ramda'
 
 import * as Survey from '@core/survey/survey'
 import * as NodeDef from '@core/survey/nodeDef'
-import * as SurveyState from '@webapp/survey/surveyState'
-import * as NodeDefUiProps from '../../surveyForm/nodeDefs/nodeDefUIProps'
 
-const AttributeSelector = props => {
+import * as NodeDefUiProps from '@webapp/loggedin/surveyViews/surveyForm/nodeDefs/nodeDefUIProps'
+
+import { useSurvey } from '@webapp/commonComponents/hooks'
+
+const AttributesSelector = (props) => {
   const {
-    nodeDefContext,
-    nodeDef,
+    canSelectAttributes,
+    filterTypes,
     lang,
+    nodeDefUuidEntity,
     nodeDefUuidsAttributes,
     onToggleAttribute,
-    filterTypes,
-    canSelectAttributes,
+    showAncestors,
     showMultipleAttributes,
   } = props
 
-  const isAttributeFn = showMultipleAttributes ? NodeDef.isAttribute : NodeDef.isSingleAttribute
-  const isVisible =
-    (isAttributeFn(nodeDef) || NodeDef.isEqual(nodeDef)(nodeDefContext)) &&
-    (R.isEmpty(filterTypes) || R.includes(NodeDef.getType(nodeDef), filterTypes))
+  const survey = useSurvey()
+  const nodeDefContext = Survey.getNodeDefByUuid(nodeDefUuidEntity)(survey)
+  const nodeDefParent = Survey.getNodeDefParent(nodeDefContext)(survey)
 
-  const nodeDefUuid = NodeDef.getUuid(nodeDef)
-  const nodeDefType = NodeDef.getType(nodeDef)
-  const isActive = R.includes(nodeDefUuid, nodeDefUuidsAttributes)
+  let childDefs = []
+  if (nodeDefContext) {
+    childDefs = NodeDef.isEntity(nodeDefContext)
+      ? Survey.getNodeDefChildren(nodeDefContext, true)(survey)
+      : [nodeDefContext] // Multiple attribute
+  }
 
-  return (
-    isVisible && (
-      <button
-        className={`btn btn-s btn-node-def${isActive ? ' active' : ''}`}
-        onClick={() => onToggleAttribute(nodeDefUuid)}
-        disabled={!canSelectAttributes}
-      >
-        {NodeDef.getLabel(nodeDef, lang)}
-        {NodeDefUiProps.getIconByType(nodeDefType)}
-      </button>
+  const AttributeSelector = ({ nodeDef }) => {
```

Comment:
I just realized you moved AttributeSelector within AttributesSelector.
Move it to its own file

Reply:
ok